spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -57,13 +62,18 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		this.binderFactory = binderFactory;
 	}
 
-	public Binding<MessageChannel> bindConsumer(MessageChannel inputChannel, String inputChannelName) {
+	public Collection<Binding<MessageChannel>> bindConsumer(MessageChannel inputChannel, String inputChannelName) {
 		String channelBindingTarget = this.channelBindingServiceProperties.getBindingDestination(inputChannelName);
```

Comment:
@ilayaperumalg I think since it is a method to bind a consumer, it is implicitly understood that we are binding on multiple destinations to an input channel. In that sense, `destination` here means `@Input`. We don't support multiple destination names for producers (`@output`)although for fanout type of use cases, it might be a conceivable idea.